### ISSUE TITLE
[Low Priority] Improve speed measurement

### ISSUE THREAD

**Issue by Anachron** [void-linux/xbps#528]
I get ridiculous speed statements when I sync from time to time.

Today I got

```
sudo xbps-install -Su
[*] Updating repository `https://repo-default.voidlinux.org/current/x86_64-repodata' ...
x86_64-repodata: 308KB [avg rate: -14GB/s]
[*] Updating repository `https://repo-default.voidlinux.org/current/nonfree/x86_64-repodata' ...
x86_64-repodata: 13KB [avg rate: 211MB/s]
```

**Comment by fvalasiad** [comment]
Well short syncs also give me crazy speed sometimes, just got `12GB/s` after syncing. Didn't manage a negative value though.

I imagine though that having lengthier interactions will actually give more accurate results, so I struggle finding reason why anyone would care.

Might look at it for fun.

**Comment by Duncaen** [comment]
Yes there is no way to actually make this reliable for small downloads. Negative is a simple bug where incomplete downloads that are begin continued will be counted wrong iirc.
